### DISCUSSION TITLE
How to access files from within an app bundle?

### DISCUSSION THREAD

**Discussion by rkennedy-argus** [progrium/darwinkit#273] [Q&A]
I've built an app with DarwinKit and placed it into a `.app` bundle, along with some ancillary data files that I need during runtime. How do I access those files? When I use [os.Getwd](https://pkg.go.dev/os#Getwd) to get the current working directory it appears to be `/`.

I assume there must be some object that I can use to get the path to where the app bundle is located so I can create the proper path to open my data file. I've skimmed the `appkit.Application` and `appkit.ApplicationDelegate` types, but didn't see anything obvious that stood out to me.

I realize this is likely more of an Apple development question than a DarwinKit one. But I figured someone here may have run into the same problem and might have the answer I'm seeking.

**Comment by rkennedy-argus** [comment]
It figures that after an hour of poking around unsuccessfully that I would post this and almost immediately stumble into the answer.

```go
bundle := foundation.Bundle_MainBundle()
path_to_file := bundle.PathForResourceOfType("foo", "txt")
```

This will return the path to the `foo.txt` file in the `MyApp.app/Contents/Resources` directory.

**Comment by progrium** [comment]
Thanks for sharing the solution!

Jeff Lindsay
*https://progrium.xyz <https://progrium.xyz>*
